io/src/openni2/openni2_device.cpp

```diff
@@ -294,9 +292,7 @@ pcl::io::openni2::OpenNI2Device::hasDepthSensor () const
 void
 pcl::io::openni2::OpenNI2Device::startIRStream ()
 {
-  boost::shared_ptr<openni::VideoStream> stream = getIRVideoStream ();
-
-  if (stream)
+  if (auto stream = getIRVideoStream ())
```

Comment:
The assignment without being enclosed in parenthesis will generate a warning if I'm not mistaken.

Reply:
Which compiler? I don't get any with GCC, also nothing on [Azure Ubuntu](https://dev.azure.com/PointCloudLibrary/pcl/_build/results?buildId=4547) (unfortunately jobs for other OSes don't build OpenNI2).

Reply:
Here's an example of what I meant
https://github.com/PointCloudLibrary/pcl/pull/3075/commits/c0a7b641e65a4f3baa208a565d63db0e16197f7a

It made me question why you're not seeing it and it's because you're invoking a ctor in these expressions, which is not an assignment.

Reply:
Thanks for the link. This was one month ago, so I don't remember :tropical_fish: 

Right, in this case there is no possibility of `=` vs. `==` mistake, i.e. the following wouldn't compile:

```cpp
if (auto stream == getIRVideoStream ()) 
```

What's your opinion, should we have parentheses nevertheless?

Reply:
> What's your opinion, should we have parentheses nevertheless?

No. It's one of the few cases of less code without readability issues.